webpack/components/extensions/HostDetails/Tabs/__tests__/tracesTab.test.js

```diff
@@ -1,127 +1,296 @@
-  // but this is closer to how the user would do it)
-  traceCheckbox.click();
-  expect(traceCheckbox.checked).toEqual(true);
-
-  const restartAppButton = getByText('Restart app');
-  // wait 50ms so that the button is enabled
-  await waitFor(() => {
-    expect(getByText('Restart app')).not.toBeDisabled();
-    restartAppButton.click();
+
+  test('Shows Enable Tracer modal', async () => {
+    const { getByText, queryByText } = renderWithRedux(<TracesTab />, renderOptions(false));
+
+    await patientlyWaitFor(() => expect(queryByText('Traces are not enabled')).toBeInTheDocument());
+    const enableTracesButton = getByText('Enable Traces');
+    enableTracesButton.click();
+    expect(queryByText('via remote execution')).toBeInTheDocument();
+
+    const cancelLink = queryByText('Cancel');
+    cancelLink.click();
+    expect(queryByText('via remote execution')).not.toBeInTheDocument();
   });
 
-  assertNockRequest(autocompleteScope);
-  assertNockRequest(resolveTracesScope);
-  assertNockRequest(scope, done); // Pass jest callback to confirm test is done
+  test('Can enable tracer via remote execution', async (done) => {
+    const jobInvocationScope = nockInstance
+      .post(jobInvocations)
+      .reply(201, mockJobInvocationStatus);
+
+    const { getByText, getByRole, queryByText }
+      = renderWithRedux(<TracesTab />, renderOptions(false));
+
+    await patientlyWaitFor(() => expect(queryByText('Traces are not enabled')).toBeInTheDocument());
+    const enableTracesButton = getByText('Enable Traces');
+    enableTracesButton.click();
+    expect(queryByText('via remote execution')).toBeVisible();
+
+    const enableTracesModalButton = getByRole('button', { name: 'Enable Tracer' });
+    fireEvent.click(enableTracesModalButton);
+    expect(queryByText('via remote execution')).not.toBeInTheDocument();
+
+    assertNockRequest(jobInvocationScope, done);
+  });
+
+  test('Can enable tracer via customized remote execution', async () => {
+    const { getByText, getByRole, queryByText }
+      = renderWithRedux(<TracesTab />, renderOptions(false));
+
+    await patientlyWaitFor(() => expect(queryByText('Traces are not enabled')).toBeInTheDocument());
+    const enableTracesButton = getByText('Enable Traces');
+    enableTracesButton.click();
+
+    const dropdown = queryByText('via remote execution');
+    await act(async () => fireEvent.click(dropdown));
+
+    const viaCustomizedRex = queryByText('via customized remote execution');
+    expect(viaCustomizedRex).toBeVisible();
+    viaCustomizedRex.click();
+    expect(queryByText('via remote execution')).not.toBeInTheDocument();
+
+    const enableTracesModalLink = getByRole('link', { name: 'Enable Tracer' });
+    expect(enableTracesModalLink)
+      .toHaveAttribute(
+        'href',
+        '/job_invocations/new?feature=katello_package_install&inputs%5Bpackage%5D=katello-host-tools-tracer&host_ids=name%20%5E%20(client.example.com)',
```

Comment:
@MariaAga For the new job invocation wizard, will we have / do we already have the capability to pass in job details via url like this?

Reply:
Yeah, should be made available in this pr: https://github.com/theforeman/foreman_remote_execution/pull/629